src/geometry_test.cxx

```diff
@@ -0,0 +1,96 @@
+import Geometry;
+//test that the shit is actually a type
+//test ray.at
+//
+
+#include <gtest/gtest.h>
+
+#include <cstdint>
+#include <vector>
+#include <iostream>
+#include <random>
+#include <chrono>
+#include <type_traits>
+
+template<Numerical T>
+class Rand_number {
+	public:
+        Rand_number() {}
+        Rand_number(double low, double high) : dist_f{low,high} {}
+        Rand_number(int low, int high) : dist_i{low,high} {}
+
+	Numerical auto operator()() {
+            if constexpr (std::is_integral_v<T>){
+                return dist_i(re); 
+            }
+            else{
+                return dist_f(re); 
+            }
+        }
+	void seed(Numerical auto s) { re.seed(s); }
+	private:
+            std::default_random_engine re;
+            std::uniform_int_distribution<> dist_i;
+            std::uniform_real_distribution<> dist_f;
+};
+
+class GeometryTest : public testing::Test {
+    protected:
+        Rand_number<float> random_float;
+        Rand_number<int> random_int;
+        
+        Point_3d<float> random_point_f;
+        Vector_3d<float> random_velocity_f;
+        Ray<float> random_ray_f;
+        
+        Point_3d<int> random_point_i;
+        Vector_3d<int> random_velocity_i;
+        Ray<int> random_ray_i;
+
+        GeometryTest()
+        {  
+            random_int = Rand_number<int>(0, 300);
+            random_float = Rand_number<float>(0, 300);
+            random_int.seed(std::chrono::system_clock::now().time_since_epoch().count());
```

Comment:
It is a bad idea to set variable seeds. Could get in the case where a run shows an error and another doesn't, which are awful to debug.